src/Symfony/Component/VarDumper/Caster/PgSqlCaster.php

```diff
@@ -0,0 +1,148 @@
+
+/**
+ * Casts pqsql resources to array representation.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class PgSqlCaster
+{
+    private static $paramCodes = array(
+        'server_encoding',
+        'client_encoding',
+        'is_superuser',
+        'session_authorization',
+        'DateStyle',
+        'TimeZone',
+        'integer_datetimes',
+    );
+
+    private static $transactionStatus = array(
+        PGSQL_TRANSACTION_IDLE => 'PGSQL_TRANSACTION_IDLE',
+        PGSQL_TRANSACTION_ACTIVE => 'PGSQL_TRANSACTION_ACTIVE',
+        PGSQL_TRANSACTION_INTRANS => 'PGSQL_TRANSACTION_INTRANS',
+        PGSQL_TRANSACTION_INERROR => 'PGSQL_TRANSACTION_INERROR',
+        PGSQL_TRANSACTION_UNKNOWN => 'PGSQL_TRANSACTION_UNKNOWN',
+    );
+
+    private static $resultStatus = array(
+        PGSQL_EMPTY_QUERY => 'PGSQL_EMPTY_QUERY',
+        PGSQL_COMMAND_OK => 'PGSQL_COMMAND_OK',
+        PGSQL_TUPLES_OK => 'PGSQL_TUPLES_OK',
+        PGSQL_COPY_OUT => 'PGSQL_COPY_OUT',
+        PGSQL_COPY_IN => 'PGSQL_COPY_IN',
+        PGSQL_BAD_RESPONSE => 'PGSQL_BAD_RESPONSE',
+        PGSQL_NONFATAL_ERROR => 'PGSQL_NONFATAL_ERROR',
+        PGSQL_FATAL_ERROR => 'PGSQL_FATAL_ERROR',
+    );
+
+    private static $diagCodes = array(
+        'severity' => PGSQL_DIAG_SEVERITY,
+        'sqlstate' => PGSQL_DIAG_SQLSTATE,
+        'message' => PGSQL_DIAG_MESSAGE_PRIMARY,
+        'detail' => PGSQL_DIAG_MESSAGE_DETAIL,
+        'hint' => PGSQL_DIAG_MESSAGE_HINT,
+        'statement position' => PGSQL_DIAG_STATEMENT_POSITION,
+        'internal position' => PGSQL_DIAG_INTERNAL_POSITION,
+        'internal query' => PGSQL_DIAG_INTERNAL_QUERY,
+        'context' => PGSQL_DIAG_CONTEXT,
+        'file' => PGSQL_DIAG_SOURCE_FILE,
+        'line' => PGSQL_DIAG_SOURCE_LINE,
+        'function' => PGSQL_DIAG_SOURCE_FUNCTION,
+    );
+
+    public static function castLargeObject($lo, array $a, Stub $stub, $isNested)
+    {
+        $a['seek position'] = pg_lo_tell($lo);
+
+        return $a;
+    }
+
+    public static function castLink($link, array $a, Stub $stub, $isNested)
+    {
+        $a['status'] = pg_connection_status($link);
+        $a['status'] = new ConstStub(PGSQL_CONNECTION_OK === $a['status'] ? 'PGSQL_CONNECTION_OK' : 'PGSQL_CONNECTION_BAD', $a['status']);
+        $a['busy'] = pg_connection_busy($link);
+
+        $a['transaction'] = pg_transaction_status($link);
+        if (isset(self::$transactionStatus[$a['transaction']])) {
+            $a['transaction'] = new ConstStub(self::$transactionStatus[$a['transaction']], $a['transaction']);
+        }
+
+        $a['pid'] = pg_get_pid($link);
+        $a['last error'] = pg_last_error($link);
+        $a['last notice'] = pg_last_notice($link);
+        $a['host'] = pg_host($link);
+        $a['port'] = pg_port($link);
+        $a['dbname'] = pg_dbname($link);
+        $a['options'] = pg_options($link);
+        $a['version'] = pg_version($link);
+
+        foreach (self::$linkParameters as $v) {
```

Comment:
Access to undeclared static property: Symfony\Component\VarDumper\Caster\PgSqlCaster::$linkParameters in /home/greg/test/symfony/src/Symfony/Component/VarDumper/Caster/PgSqlCaster.php on line 94